### PR TITLE
Don't terminate topics on signals

### PR DESCRIPTION
**Pull request by peterebden**
We still want to be able to send messages out after receiving a signal (if we're in the middle of processing a build) but don't want to receive any more.

I can't see a particularly nice way of hooking it up on the API server side but honestly not sure it's especially necessary (the [gcppubsub implementation](https://github.com/google/go-cloud/blob/2330ef456904e4ed497045a20bbccf61d43f8f5c/pubsub/gcppubsub/gcppubsub.go#L369) doesn't seem to do much with it anyway).

### CODE REVIEW

File: mettle/worker/worker.go [hunk 1]
```diff
@@ -151,6 +148,7 @@ func runForever(instanceName, requestQueue, responseQueue, name, storage, dir, c
 	if err != nil {
 		return err
 	}
+	defer w.Shutdown()
```
Comment (Hamishpk) [inline mettle/worker/worker.go:151]: nit: Would it be worth expanding on the name here to make it clear what's being shut down? At first look my assumption was it shutdown the worker (which obviously didn't make sense).

File: mettle/worker/worker.go [hunk 2]
```diff
@@ -122,19 +122,16 @@ func RunOne(instanceName, name, storage, dir, cacheDir, sandbox, altSandbox, tok
...
-		time.Sleep(50 * time.Millisecond) // this is dodgy obvs
-		b, _ := proto.Marshal(&pb.ExecuteRequest{
-			InstanceName: instanceName,
-			ActionDigest: digest,
-		})
-		log.Notice("Sending request to build %s...", digest.Hash)
-		if err := topic.Send(context.Background(), &pubsub.Message{Body: b}); err != nil {
-			log.Fatalf("Failed to submit job to internal queue: %s", err)
-		}
-		log.Notice("Sent request to build %s", digest.Hash)
```
Comment (Hamishpk) [inline mettle/worker/worker.go:136]: It might be worth keeping this log line and the one above.

**Review by Hamishpk — APPROVED** [review]
A couple of nits but looks good to me!

File: mettle/worker/worker.go (hunk 1 above)
Comment (peterebden) [reply mettle/worker/worker.go]: Yeah fair enough, I was thinking of it as being kind of generic but probably better just to be explicit.

File: mettle/worker/worker.go (hunk 2 above)
Comment (peterebden) [reply mettle/worker/worker.go:136]: Yeah fair, I think it is less of a deal now it's synchronous but doesn't hurt to keep them.

**Pull request merged**